### ISSUE TITLE
add relative slot input to distance node

### ISSUE THREAD

**Issue by epicEaston197** [Neos-Metaverse/NeosPublic#3809]
### Is your tweak request related to a problem? Please describe.

this would basically get the distance between two points relative to a slot instead of the node's position
![image](https://user-images.githubusercontent.com/76523692/207926334-b885a856-4b1c-4453-9f29-f45d78ca8721.png)

### Describe what would you like tweaked

I would like a slot input on the distance node to get the distance between two points relative to the slot

### Describe alternatives you've considered

just move the distance note

**Comment by LexiBasilisk** [comment]
I'm trying to understand what you're trying to accomplish here?

You already can use the global transform nodes to get the position of a desired slot to compare against using the distance node.

Having more context about what you're trying to do would help, but I don't feel this is an appropriate addition to the distance node based on what you've shown so far. If you know what the slot is, you can already do this without this addition.

**Comment by epicEaston197** [comment]
> I'm trying to understand what you're trying to accomplish here?
> 
> You already can use the global transform nodes to get the position of a desired slot to compare against using the distance node.
> 
> Having more context about what you're trying to do would help, but I don't feel this is an appropriate addition to the distance node based on what you've shown so far. If you know what the slot is, you can already do this without this addition.

basically if I move this room it will not move the distance node and I can't parent the Logics to the room or else the Logix won't work because they will be disabled I am asking for a slot input so the distance calculation is relative to the room 

similar to nearest user head the note will detect the closest user to the node but there is a slot input to make it relative to a slot 

I hope it makes it more clear of what I am asking

**Comment by LexiBasilisk** [comment]
Is it a fixed point relative to the room? If so, you could just make a slot at that point in space relative to the room's position (or just use the room's slot), then use the global transform node to get the position of that slot relative to the local user's hip position.

**Comment by epicEaston197** [comment]
> Is it a fixed point relative to the room? If so, you could just make a slot at that point in space relative to the room's position (or just use the room's slot), then use the global transform node to get the position of that slot relative to the local user's hip position.

yeah I could do that but I think it will be more consistent to have a slot input as I would have expected it to have that it also makes it more consistent with the other distance-based nodes like nearest user head and hand

this feature doesn't really matter as I don't move the rooms around too much but would definitely be convenient

**Comment by ProbablePrime** [comment]
I don't think we'd be adding this.

Many of our float/int nodes are basically "pure maths" they don't make any assumptions about slots, spaces etc. Think about +-/* etc. and there's most certainly a desire to keep them like this. It makes it easier in the long run to do many mathematical operations. 

Distance is only "sometimes" about slots. it can also just be about pure distances(say gun/ballistics/rocket calculations/ai etc).

Therefore not making assumptions and tying this node to the slot system is by design.

Take a look again into the transform folder, those nodes are designed for moving between spaces. What you're meant to do is to get your spaces all synced up using the transform nodes and then do maths on them.

To put it another way that might help, if we added slot inputs to distance, what's the next node you'd ask for them to also be added to? +-/*? Max, IsBetween etc.

At that point should the Transform folder then exist?

Generically with LogiX, we're trying to give you the tools to build as many complicated things as we can and part of doing that is keeping things really open and avoiding too much coupling. Distance is a purely mathematical operation and I think it should stay there.

Oh and finally, just adding one slot input isn't enough. you'll need to align both objects to the correct space for an accurate distance. This is usually done with Global Transform. But if you really wanted slot inputs on the distance node you'd need two so that we can figure out the translation between the spaces.
